### PR TITLE
Added PR labels

### PR DESCRIPTION
**Pull request by WalshyDev**
## Description
Made this action forever ago, was meant to add it in Hacktoberfest but... yeah.

Anyway, this adds labels to PRs based on the branch prefix. This is good for categorisation and helpful for people like myself for which PRs to look at first. We should add more labels but I thought these were some basic starting ones.

## Changes
* new workflow file to add labels

## Related Issues
N/A

## Checklist

- [X] I have fully tested the proposed changes and promise that they will not break everything into chaos.
- [ ] I have also tested the proposed changes in combination with various popular addons and can confirm my changes do not break them.
- [ ] I followed the existing code standards and didn't mess up the formatting.
- [ ] I did my best to add documentation to any public classes or methods I added.
- [ ] I have added `Nonnull` and `Nullable` annotations to my methods to indicate their behaviour for null values
- [ ] I added sufficient Unit Tests to cover my code.

### CODE REVIEW

**Review by svr333 — DISMISSED** [review]
Nice addition, wont trigger many if any false positives
Im all for it

**Review by Sfiguz7 — APPROVED** [review]
Mr Blob forced me at gunpoint, please he has my children




LGTM

File: .github/workflows/pr-labels.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+name: PR Labeler
+on:
+  pull_request:
+    types: [opened]
+
+jobs:
+  pr-labeler:
+    runs-on: ubuntu-latest
```
Comment (TheBusyBiscuit) [inline .github/workflows/pr-labels.yml:8]: Missing job name.

File: .github/workflows/pr-labels.yml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    steps:
+      - uses: WalshyDev/pr-labels@v1
```
Comment (TheBusyBiscuit) [inline .github/workflows/pr-labels.yml:10]: Missing name for this step

File: .github/workflows/pr-labels.yml [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+        with:
+          token: "${{ secrets.GITHUB_TOKEN }}"
```
Comment (TheBusyBiscuit) [inline .github/workflows/pr-labels.yml:12]: We can use the token for @TheBusyBot here.

```suggestion
          token: "${{ secrets.ACCESS_TOKEN }}"
```

File: .github/workflows/pr-labels.yml [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+          feature: 'New Feature'
```
Comment (TheBusyBiscuit) [inline .github/workflows/pr-labels.yml:13]: emoji? :eyes:

File: .github/workflows/pr-labels.yml [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+          fix: '🐞 Bug Fix'
+          chore: 'Chore'
```
Comment (TheBusyBiscuit) [inline .github/workflows/pr-labels.yml:15]: emoji? :eyes:

File: .github/workflows/pr-labels.yml [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+          perf: '🛠️ Performance Improvement'
+          performance: '🛠️ Performance Improvement'
```
Comment (TheBusyBiscuit) [inline .github/workflows/pr-labels.yml:17]: Redundant. I think we should remove one of those, or we could approach it with a regular expression.

**Review by TheBusyBiscuit — CHANGES_REQUESTED** [review]
I'll be honest and say I am not really a big fan of this, seems quite unnecessary and I definitely would veto against this being a requirement for PRs.

I am not a big fan of having too many labels, but if you guys find this helpful, sure, we can always manually label PRs, they come in rare enough for that to be not too much of a stretch.

**Comment by TheBusyBiscuit** [comment]
Would be awesome if the changes will be addressed, I took the liberty to add labels for it already.

* "🧹 Chores"
* "💡 Performance Optimization"
* "🎈 Feature"
* "✨ Fix"

**Review by TheBusyBiscuit — COMMENTED** [review]
I fixed the given issues myself now.
One suggestion I still have would be have an output for this action, something like "applied-label".
Then we could do a null check on this and perhaps make TheBusyBot leave a comment like "Hey, we recommend you to use this branch naming convention..."

**Comment by WalshyDev** [comment]
Sure, looking into that now

**Comment by WalshyDev** [comment]
image

Yeah :sunglasses:

**Comment by WalshyDev** [comment]
Pushed v1.1 and updated the workflow file

Example of usage is on the release page: https://github.com/WalshyDev/pr-labels/releases/tag/v1.1

**Review by TheBusyBiscuit — APPROVED** [review]

**Pull request merged**